### PR TITLE
Add snap to grid option

### PR DESCRIPTION
**Pull request by FaultyFunctions**
This is super useful for organizing better node structures. This also works with all the align functions as well! Let me know if anything needs to be changed!

### CODE REVIEW

**Comment by JujuAdams** [comment]
Excellent, been wanting this one for ages

**Comment by blurymind** [comment]
Looks good to me

File: src/js/classes/app.js [hunk 1]
```diff
@@ -89,6 +89,7 @@ export var App = function(name, version) {
   this.editingPath = ko.observable(null);
   this.$searchField = $('.search-field');
   this.isEditorInPreviewMode = false;
+  this.gridSize = 40;
```
Comment (blurymind) [inline src/js/classes/app.js:92]: It would be nice if you also expose this to the settings

**Comment by FaultyFunctions** [comment]
Yeah I'll expose `app.gridSize` to the settings dialogue. I'm also working on adding a grid element to the a new grid canvas. I'll push an update once I get it completed! Thanks!

**Comment by FaultyFunctions** [comment]
Okay! This took a long time but I learned a lot about the HTML canvas property and learned some coordinate math as well. I especially am excited to show you guys the blueprint theme since both the subdivision width and subdivision amount scale to the canvas when zooming. Let me know what you guys think and if any changes are needed!

**Comment by blurymind** [comment]
Fantastic work on this! 👍 😄 
It's perfect imo

**Comment by blurymind** [comment]
Merged! 🎉 
Thank you for implementing this feature. It has been requested alot in the past and your implementation is very clean

**Pull request merged**